src/Processors/Chunk.h

```diff
@@ -114,6 +120,17 @@ class Chunk
 
 using Chunks = std::vector<Chunk>;
 
+/// To carry part level if chunk is produced by a merge tree source
+class MergeTreePartLevelInfo : public ChunkInfo
```

Comment:
This class is better to declare whenever it would be used.  (Somewhere inside `src/Processors/Merges/Algorithms/`)